### ISSUE TITLE
Write proper introduction, documentation, and examples

### ISSUE THREAD

**Issue by tsugabloom** [heyheyhello/haptic#6]
There's no proper introduction to Haptic at all right now. I've put off writing documentation for a very long time. I need to also _delete_ the examples/ folder.

It's blocked on porting styletakeout.macro from Babel to Acorn so I can have CSS-in-JS. I've never said this but, while not part of the library directly, CSS-in-JS is an important part of Haptic's values along with not being tied to heavy tools like Webpack/Babel.

It's an important part of showcasing examples; many reactive libraries show an example using React/Preact for a similar reason. It needs to be tangible and appealing in ways that my sandbox examples/ folder isn't.

Then I'll create a dedicated repo for examples (interactive, styled, literate programming) to introduce features.
